stellargraph/mapper/sampled_node_generators.py

```diff
@@ -495,6 +496,14 @@ def sample_features(self, head_nodes, batch_num):
 
         return batch_feats
 
+    def default_corrupt_input_index_groups(self):
+        # every sample of a given node type can be grouped together
+        indices_per_nt = defaultdict(list)
+        for tensor_idx, (nt, _) in enumerate(self._sampling_schema[0]):
```

Comment:
not related to this PR (and so doesn't need to be fixed), but it seems unecessary that `self._sampling_schema[0]` is a list of lists. From what I can tell the outer list always has one element and we only access the first element of the outer list anyway.

Reply:
I think it has two elements for link prediction, where each element represents the input tensor layout for each node of the edge.